Automattic/wp-calypso
# Checkout: Convert all stripe redirect payment processors to use automated responses

#### Changes proposed in this Pull Request

This is a follow-up to https://github.com/Automattic/wp-calypso/pull/46337 which converts all the Stripe redirect payment processors to use the new system.

It also splits the WeChat payment processor into its own processor because on desktop it has quite a different behavior. That behavior remains mostly encoded into the WeChat button, but in order to use the `onClick` handler, we need to tell the processor function to do nothing with the response. (I also made a mistake in the original PR where I assumed that a NOOP response would be sufficient for a payment method to do its own success/failure handling, but it's not because we also need the transaction response, so this PR replaces NOOP with MANUAL.)

#### Testing instructions

Since all these payment methods use the same payment processor function, they will all need to be tested in this one PR.

- WeChat: see https://github.com/Automattic/wp-calypso/pull/43765
- NetBanking: see https://github.com/Automattic/wp-calypso/pull/44952
- IdWallet: see https://github.com/Automattic/wp-calypso/pull/45257
- EbanxTef: see https://github.com/Automattic/wp-calypso/pull/44752
- Sofort: https://github.com/Automattic/wp-calypso/pull/43710
- P24: https://github.com/Automattic/wp-calypso/pull/43744
- Ideal: https://github.com/Automattic/wp-calypso/pull/43162
- Giropay: https://github.com/Automattic/wp-calypso/pull/43598
- Eps: see https://github.com/Automattic/wp-calypso/pull/43734
- BanContact: see https://github.com/Automattic/wp-calypso/pull/43651
- Alipay: see https://github.com/Automattic/wp-calypso/pull/43653
